### ISSUE TITLE
createClient doesn't work with URL on port 80 or 443 since 1.0.0

### ISSUE THREAD

**Issue by martoche** [ClickHouse/clickhouse-js#258]
### Describe the bug

Can't connect to a Clickhouse server that listens on port 443 (or when a reverse proxy on port 443 is used to forward requests to the Clickhouse server on any other port).

The following code :
```js
const clickhouseClient = createClient({
  url: "https://localhost:443",
  pathname: "clickhouse",
  database: "apm",
})
```

Throws the error: `Uncaught Error: ClickHouse URL must contain a valid port number.`

This is because of these lines: https://github.com/ClickHouse/clickhouse-js/blob/64b5f8128cef4a868509aec3ce3879c7c9ca841a/packages/client-common/src/config.ts#L286-L288

The condition in the `if` is incorrect: when the URL scheme is "http" and the port is explicitely set as "80", then `new URL("http://localhost:80").port` is equal to the empty string. Similarily, `new URL("https://localhost:443").port === ""`.

This is a regression of version [1.0.0](https://github.com/ClickHouse/clickhouse-js/releases/tag/1.0.0), it works in 0.3.1.

#### Environment
* Client version: 1.0.0 (web)

**Comment by slvrtrn** [comment]
On it, thanks for the report.

**Comment by slvrtrn** [comment]
@martoche, it should be fixed in 1.0.1.
Thanks for the report!
